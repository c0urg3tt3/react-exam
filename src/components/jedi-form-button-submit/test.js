import React from 'react'
import { shallow } from 'enzyme'

import JediFormButtonSubmit from './component'

describe('[component] JediFormButtonSubmit', () => {
  it('should render component', () => {
    const wrapper = shallow(<JediFormButtonSubmit />)

    expect(wrapper.exists()).toEqual(true)
    expect(wrapper.is('button')).toEqual(true)
    expect(wrapper.hasClass('jedi-form-button-submit')).toEqual(true)
    expect(wrapper.prop('type')).toBe('submit')
    expect(wrapper.prop('title')).toBe('[disabled] Submit your padawan to the Council')
    expect(wrapper.prop('disabled')).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should enable the button', () => {
    const wrapper = shallow(<JediFormButtonSubmit />)

    expect(wrapper.prop('disabled')).toBe(true)
    expect(wrapper.prop('title')).toBe('[disabled] Submit your padawan to the Council')
    expect(wrapper.html()).toMatchSnapshot()

    wrapper.setProps({isDisabled: false})

    expect(wrapper.prop('disabled')).toBe(false)
    expect(wrapper.prop('title')).toBe('Submit your padawan to the Council')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should accept child', () => {
    const wrapper = shallow(
      <JediFormButtonSubmit>
        <img alt="child"/>
      </JediFormButtonSubmit>
    )

    expect(wrapper.find('img')).toHaveLength(1)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should accept children', () => {
    const wrapper = shallow(
      <JediFormButtonSubmit>
        <img alt="child"/>
        <img alt="child"/>
      </JediFormButtonSubmit>
    )

    expect(wrapper.find('img')).toHaveLength(2)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
