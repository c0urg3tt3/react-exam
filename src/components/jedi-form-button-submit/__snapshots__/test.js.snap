// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[component] JediFormButtonSubmit should accept child 1`] = `"<button class=\\"jedi-form-button-submit\\" type=\\"submit\\" title=\\"[disabled] Submit your padawan to the Council\\" disabled=\\"\\"><img alt=\\"child\\"/></button>"`;

exports[`[component] JediFormButtonSubmit should accept children 1`] = `"<button class=\\"jedi-form-button-submit\\" type=\\"submit\\" title=\\"[disabled] Submit your padawan to the Council\\" disabled=\\"\\"><img alt=\\"child\\"/><img alt=\\"child\\"/></button>"`;

exports[`[component] JediFormButtonSubmit should enable the button 1`] = `"<button class=\\"jedi-form-button-submit\\" type=\\"submit\\" title=\\"[disabled] Submit your padawan to the Council\\" disabled=\\"\\"></button>"`;

exports[`[component] JediFormButtonSubmit should enable the button 2`] = `"<button class=\\"jedi-form-button-submit\\" type=\\"submit\\" title=\\"Submit your padawan to the Council\\"></button>"`;

exports[`[component] JediFormButtonSubmit should render component 1`] = `"<button class=\\"jedi-form-button-submit\\" type=\\"submit\\" title=\\"[disabled] Submit your padawan to the Council\\" disabled=\\"\\"></button>"`;
